App/authPopup.js

```diff
@@ -14,11 +14,21 @@ function selectAccount () {
 
     const currentAccounts = myMSALObj.getAllAccounts();
 
-    if (currentAccounts.length === 0) {
+    if (!currentAccounts || currentAccounts.length < 1) {
         return;
-    } else if (currentAccounts.length > 1) {
-        // Add your account choosing logic here
-        console.log("Multiple accounts detected.");
+    } else if (currentAccounts.length === Object.entries(b2cPolicies.names).length) {
+        
+        /**
+         * Due to the way MSAL caches account objects, the auth response from initiating a user-flow
+         * is cached as a new account. Here we make sure we are selecting the account with homeAccountId
+         * that contains the sign-up/sign-in user-flow.
+         */
+        const account = currentAccounts.find(account => 
+            account.homeAccountId.toUpperCase().includes(b2cPolicies.names.signUpSignIn.toUpperCase()));
```

Comment:
Ah, nice idea! If you want to go this route I would also recommend checking `clientId` and authority match and then also that there aren't more than 1 user signed into this authority/policy.

Another way to do this would be to track the `homeAccountId` after retrieving tokens and use the `getAccountByHomeId` API or use the `setActiveAccount`/`getActiveAccount` APIs.

Reply:
Either way, this is a great improvement over logging out and logging back in again!

Reply:
@derisen Is this a part of the improved B2C policy experience or an enhancement we added?

Reply:
@tnorling thanks a lot for the help! I was under the impression that `set/getActiveAccount` is Angular specific. To keep track of the `homeAccountId`, would it be safe to use the browser storage? I wasn't sure to go that route. Failing that, I've come up with something like this:

```javascript
function selectAccount() {
    /**
     * See here for more info on account retrieval: 
     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/docs/Accounts.md
     */

    const currentAccounts = myMSALObj.getAllAccounts();

    if (currentAccounts.length < 1) {
        return;
    } else if (currentAccounts.length > 1) {

        /**
         * Due to the way MSAL caches account objects, the auth response from initiating a user-flow
         * is cached as a new account. Here we make sure we are selecting the account with homeAccountId
         * that contains the sign-up/sign-in user-flow, as this is the default flow we initially signed-in with.
         */
        const accounts = currentAccounts.filter(account =>
            account.homeAccountId.toUpperCase().includes(b2cPolicies.names.signUpSignIn.toUpperCase())
            &&
            account.idTokenClaims.iss.toUpperCase().includes(b2cPolicies.authorityDomain.toUpperCase())
            &&
            account.idTokenClaims.aud === msalConfig.auth.clientId 
            );

        if (accounts.length > 1) {
            if (accounts.every(account => account.localAccountId === accounts[0].localAccountId)) {
                // All accounts belong to the same user
                setAccount(accounts[0]);
            } else {
                // Multiple users detected
                signOut();
            };
        } else if (accounts.length === 1) {
            setAccount(accounts[0]);
        }

    } else if (currentAccounts.length === 1) {
        setAccount(currentAccounts[0]);
    }
}
```

@sameerag this is just an enhancement. Forcing to sign out the user wasn't the best experience.